src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -0,0 +1,275 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_integration.h>
+#include <tuple>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+/// Numerical integration algorithms
+namespace integrate {
+
+/// The integration result is returned in this format.
+struct Result {
+  double value;
+  double error;
+};
+
+namespace detail {
+template <typename... Args>
+class GslQuadAdaptiveImpl;
+
+template <typename Args, typename ArgsIndices =
+                             std::make_index_sequence<tmpl::size<Args>::value>>
+struct ExpandTupleImpl;
+
+template <typename... Args, size_t... ArgsIndices>
+struct ExpandTupleImpl<tmpl::list<Args...>,
+                       std::index_sequence<ArgsIndices...>> {
+  template <typename FunctionType>
+  static double apply(FunctionType&& function, const double x,
+                      const std::tuple<Args...>& args) {
+    return (*function)(x, std::get<ArgsIndices>(args)...);
+  }
+};
+
+template <typename... Args>
+double integrand(double x, void* params) {
+  auto args = reinterpret_cast<GslQuadAdaptiveImpl<Args...>*>(params);
+  return ExpandTupleImpl<tmpl::list<Args...>>::apply(args->function_, x,
```

Comment:
Since #2083 was just merged you can use it here:
- Include `Utilities/Tuple.hpp`
- Replace lines 46-47 with:
  ```cpp
  return cpp17::apply(
      [&args, &x](const Args&... tuple_args) {
        return (args->function_)(x, tuple_args...);
      },
      args->args_);
  ```
- Delete the `ExpandTupleImpl`